mesonbuild/modules/python.py

```diff
@@ -620,6 +622,50 @@ def __init__(self, interpreter: 'Interpreter') -> None:
             'find_installation': self.find_installation,
         })
 
+    def get_devenv(self, state: 'ModuleState', install_data: 'InstallData') -> T.Optional[EnvironmentVariables]:
+        python = PythonExternalProgram('python3', mesonlib.python_command)
+        if not python.found() or not python.sanity(state):
+            return None
```

Comment:
This will be wrong as a general rule whenever find_installation has a real argument.

More generally it seems entirely backwards as a matter of course -- we have this information in the found installation, we know inside the found installation whether we ever invoke installation rules, and we're guaranteed to know exactly where it is installed without using *very* gnarly code to guess at it.

Reply:
I thought about doing this in `install_sources()` and `extension_module()` which would be nicer of course. However I was afraid that people would be using a lot `install_subdir(..., install_dir: python.get_install_dir())` which would be missed.

That said, I could mix both and use install_data only for the subdir, which is a much smaller list to iterate. Would that be better?

Reply:
Actually, I changed my mind. I'll setup PYTHONPATH when calling `install_sources()` and `extension_module()` and if user installs python modules manually using `install_data()` or `install_subdir()` they can always setup PYTHONPATH themself as well:
```meson
env = environment()
env.prepend('PYTHONPATH', meson.current_source_dir())
meson.add_devenv(env)
```

@eli-schwartz would that be better?

Reply:
Made that change now.